test/extended/registry/signature.go

```diff
@@ -0,0 +1,120 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("obtaining bearer token for the test user")
+		user := oc.Username()
+		token, err := oc.Run("whoami").Args("-t").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("granting the image-signer role to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-signer", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// TODO: The test user needs to be able to unlink the /dev/random which is owned by a
+		// root. This cannot be done during image build time because the /dev is plugged into
+		// container after it starts. This SCC could be avoided in the future when /dev/random
+		// issue is fixed in Docker.
+		g.By("granting the anyuid scc to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "anyuid", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("preparing the image stream where the signed image will be pushed")
+		_, err = oc.Run("create").Args("imagestream", "signed").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("granting the image-auditor role to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-auditor", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		pod, err := exutil.NewPodExecutor(oc, "sign-and-push", signerImage)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Generate GPG key
+		// Note that we need to replace the /dev/random with /dev/urandom to get more entropy
+		// into container so we can successfully generate the GPG keypair.
+		g.By("creating dummy GPG key")
+		out, err := pod.Exec("rm -f /dev/random; ln -sf /dev/urandom /dev/random && " +
+			"GNUPGHOME=/var/lib/origin/gnupg gpg2 --batch --gen-key dummy_key.conf")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("keyring `/var/lib/origin/gnupg/secring.gpg' created"))
+
+		// Create kubeconfig for skopeo
+		g.By("logging as a test user")
+		out, err = pod.Exec("oc login https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT --token=" + token + " --certificate-authority=/run/secrets/kubernetes.io/serviceaccount/ca.crt")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("Logged in"))
+
+		// Sign and copy the origin-pod image into targed image stream tag
+		// TODO: Fix skopeo to pickup the Kubernetes environment variables (remove the $KUBERNETES_MASTER)
+		g.By("signing the origin-pod:latest image and pushing it into openshift registry")
+		_, err = pod.Exec("KUBERNETES_MASTER=https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT GNUPGHOME=/var/lib/origin/gnupg " +
+			"skopeo --debug --tls-verify=false copy --sign-by joe@foo.bar --dest-creds " + user + ":" + token + " --dest-tls-verify=false docker://docker.io/openshift/origin-pod:latest atomic:" + signedImage)
```

Comment:
@miminar i have no idea how to plumb the docker socket to skopeo in container... i guess it will make the test pretty complicated...